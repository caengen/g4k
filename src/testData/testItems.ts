import { Item } from "../types";

export const testAItems = [
  {
    id: "1",
    name: "A",
    value: 50
  },
  {
    id: "1a",
    name: "A",
    value: 50
  },
  {
    id: "1b",
    name: "A",
    value: 50
  },
];

export const testBItems = [
  {
    id: "2",
    name: "B",
    value: 30
  },
  {
    id: "2a",
    name: "B",
    value: 30
  }
];

export const testCItems = [
  {
    id: "3",
    name: "C",
    value: 20
  },
  {
    id: "3a",
    name: "C",
    value: 20
  },
  {
    id: "3b",
    name: "C",
    value: 20
  },
];

export const testDItems = [
  {
    id: "4",
    name: "D",
    value: 15
  },
  {
    id: "4a",
    name: "D",
    value: 15
  },
  {
    id: "4b",
    name: "D",
    value: 15
  },
];

export const testItems = [
  ...testAItems,
  ...testBItems,
  ...testCItems,
  ...testDItems
];