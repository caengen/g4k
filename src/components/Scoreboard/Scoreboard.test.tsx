import React from "react";
import { shallow } from "enzyme";
import { Scoreboard } from ".";
import { testItems } from "../../testData";

it("renders correctly", () => {
  const wrapper = shallow(<Scoreboard items={testItems} />);
  expect(wrapper).toMatchSnapshot();
});