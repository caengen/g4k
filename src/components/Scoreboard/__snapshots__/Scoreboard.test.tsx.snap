// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.table>
    <styled.thead>
      <styled.tr>
        <styled.th>
          Item
        </styled.th>
        <styled.th>
          Qty
        </styled.th>
        <styled.th>
          Score
        </styled.th>
      </styled.tr>
    </styled.thead>
    <styled.tbody>
      <styled.tr
        key="A"
      >
        <styled.th>
          <ItemCard
            item={
              Object {
                "id": "1",
                "name": "A",
                "value": 50,
              }
            }
            size={0}
          />
        </styled.th>
        <styled.th>
          3
        </styled.th>
        <styled.th>
          200
        </styled.th>
      </styled.tr>
      <styled.tr
        key="B"
      >
        <styled.th>
          <ItemCard
            item={
              Object {
                "id": "2",
                "name": "B",
                "value": 30,
              }
            }
            size={0}
          />
        </styled.th>
        <styled.th>
          2
        </styled.th>
        <styled.th>
          90
        </styled.th>
      </styled.tr>
      <styled.tr
        key="C"
      >
        <styled.th>
          <ItemCard
            item={
              Object {
                "id": "3",
                "name": "C",
                "value": 20,
              }
            }
            size={0}
          />
        </styled.th>
        <styled.th>
          3
        </styled.th>
        <styled.th>
          60
        </styled.th>
      </styled.tr>
      <styled.tr
        key="D"
      >
        <styled.th>
          <ItemCard
            item={
              Object {
                "id": "4",
                "name": "D",
                "value": 15,
              }
            }
            size={0}
          />
        </styled.th>
        <styled.th>
          3
        </styled.th>
        <styled.th>
          45
        </styled.th>
      </styled.tr>
    </styled.tbody>
  </styled.table>
  <styled.div>
    Bonus 
    80
  </styled.div>
  <styled.div>
    <styled.div>
      <span>
        Total
      </span>
      <span>
        395
      </span>
    </styled.div>
    <styled.button>
      <strong>
        New game
      </strong>
    </styled.button>
  </styled.div>
</styled.div>
`;
